unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarShape.cs

```diff
@@ -113,11 +113,11 @@ public override IEnumerator ApplyChanges(BaseModel newModel)
             EnablePassport();
 
             KernelConfig.i.EnsureConfigInitialized()
-                        .Then(config =>
-                        {
-                            if(config.features.enableAvatarLODs)
-                                avatarRenderer.InitializeImpostor();
-                        });
+                .Then(config =>
```

Comment:
Isn't this prone to race conditions because kernel initially sents the `enableAvatarLODs` as false until it reads the feature flag?

Reply:
If that happens, the `Destroy()` will be called anyways but the resources will be null and nothing will happen. (`Destroy` don't do anything with null arguments).

Reply:
I'm worried about:
1) Kernel is retrieving FeatureFlags, sends -> `enableAvatarLODs = false` in the meantime.
2) We go trough the `EnsureInitialization`
3) `avatarRenderer.InitializeImpostor()` is NOT called.
4) Kernel retrieves FeatureFlags, sends -> `enableAvatarLODs = true`.
5) AvatarShape doesn't react to it.